panels/site_sections/attractions_admin.py

```diff
@@ -0,0 +1,330 @@
+from uber.common import *
+from panels.models.attraction import *
+
+
+@all_renderable(c.STUFF)
+class Root:
+    def index(self, session, filtered=False, message='', **params):
+        if filtered:
+            admin_account_id = cherrypy.session['account_id']
+            attraction_filter = [Attraction.owner_id == admin_account_id]
+        else:
+            attraction_filter = []
+
+        attractions = session.query(Attraction).filter(*attraction_filter) \
+            .options(
+                subqueryload(Attraction.department),
+                subqueryload(Attraction.owner)
+                    .subqueryload(AdminAccount.attendee)) \
+            .order_by(Attraction.name).all()
+
+        return {
+            'filtered': filtered,
+            'message': message,
+            'attractions': attractions
+        }
+
+    def form(self, session, message='', **params):
+        attraction_id = params.get('id')
+        if not attraction_id or attraction_id == 'None':
+            raise HTTPRedirect('index')
+
+        if cherrypy.request.method == 'POST':
+            if 'notifications' in params:
+                ns = listify(params.get('notifications', []))
+                params['notifications'] = [int(n) for n in ns if n != '']
+
+            attraction = session.attraction(
+                params,
+                bools=Attraction.all_bools,
+                checkgroups=Attraction.all_checkgroups)
+            message = check(attraction)
+            if not message:
+                if not attraction.department_id:
+                    attraction.department_id = None
+                session.add(attraction)
+                raise HTTPRedirect(
+                    'form?id={}&message={}',
+                    attraction.id,
+                    '{} updated successfully'.format(attraction.name))
+        else:
+            attraction = session.query(Attraction) \
+                .filter_by(id=attraction_id) \
+                .options(
+                    subqueryload(Attraction.department),
```

Comment:
This is just to double-check, but if the attraction doesn't have a department set, this subqueryload won't cause any major problems, right?

Reply:
Should be okay